### PR TITLE
feat!: upgrade ci, image, packages, audit

### PR DESCRIPTION
**Pull request by kleyow**
- Upgrade to node v16
- Update packages
- Update ci with dep/audit checks
- Update integration tests to add some funds to payer due to `central-ledger` v14.0.0 breaking change.
- Lint

Breaking change: Docker image folder structure no longer  uses opt/central-settlement/ as the work directory and now uses /opt/app/. Update any mount file paths accordingly.

Major version bump since this is a big upgrade.

### CODE REVIEW

**Comment by kleyow** [comment]
```
NAME: moja3
LAST DEPLOYED: Tue Jul 12 12:25:55 2022
NAMESPACE: moja3
STATUS: deployed
REVISION: 1
TEST SUITE:     moja3-test
Last Started:   Tue Jul 12 12:39:23 2022
Last Completed: Tue Jul 12 12:39:36 2022
Phase:          Succeeded
TEST SUITE:     moja3-test
Last Started:   Tue Jul 12 12:39:37 2022
Last Completed: Tue Jul 12 12:39:49 2022
Phase:          Succeeded
TEST SUITE:     moja3-ml-ttk-test-setup
Last Started:   Tue Jul 12 12:34:55 2022
Last Completed: Tue Jul 12 12:35:47 2022
Phase:          Succeeded
TEST SUITE:     moja3-ml-ttk-test-val-bulk
Last Started:   Tue Jul 12 12:38:54 2022
Last Completed: Tue Jul 12 12:39:23 2022
Phase:          Succeeded
TEST SUITE:     moja3-ml-ttk-test-val-gp
Last Started:   Tue Jul 12 12:35:47 2022
Last Completed: Tue Jul 12 12:38:53 2022
Phase:          Succeeded
```

File: .ncurc.yml [hunk 1]
```diff
@@ -1,4 +0,0 @@
-{
```
Comment (mdebarros) [inline .ncurc.yml:1]: Please add the following

```yaml
## Add a TODO comment indicating the reason for each rejected dependency upgrade added to this list, and what should be done to resolve it (i.e. handle it through a story, etc).
reject: [
  // @hapi/catbox-memory v6 and onwards breaks tests. Needs investigation.
  "@hapi/catbox-memory"
  # TODO: v6+ (ref: https://github.com/sindresorhus/get-port/releases/tag/v6.0.0) is an ESM library and thus not compatible with CommonJS. Future story needed to resolve.
  "get-port"
]
```

And  remove `.ncurc.js`

File: package.json [hunk 2]
```diff
@@ -137,10 +113,12 @@
...
         "test:int:narrow": "jest --config './jest.integration.config.js' --coverage --runInBand --testMatch '**/test/int/**/*.(test|spec).js'",
-        "audit:resolve": "SHELL=sh resolve-audit --production",
-        "audit:check": "SHELL=sh check-audit --production",
+        "audit:resolve": "resolve-audit --production",
+        "audit:check": "check-audit --production",
         "dep:check": "npx ncu -e 2",
-        "dep:update": "npx ncu -u"
+        "dep:update": "npx ncu -u",
+        "release": "standard-version --releaseCommitMessageFormat 'chore(release): {{currentTag}} [skip ci]'",
+        "snapshot": "standard-version --no-verify --skip.changelog --prerelease snapshot --releaseCommitMessageFormat 'chore(snapshot): {{currentTag}}'"
```
Comment (mdebarros) [inline package.json:121]: We are missing the `standard-version` config from here: https://github.com/mojaloop/central-ledger/blob/master/package.json#L150

Don't forget to add the `replace` dev dependency.

File: .circleci/config.yml [hunk 3]
```diff
@@ -118,63 +167,37 @@ jobs:
...
-          command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ];
-            then
-                echo "Sending lcov.info to SonarQube..."
-                aws s3 cp coverage/lcov.info $AWS_S3_DIR_SONARQUBE/central-settlement/lcov.info
-            else
-                echo "Not a release (env CIRCLE_BRANCH != 'master'), skipping sending lcov.info to SonarQube."
-            fi
-
-  test-integration:
```
Comment (mdebarros) [inline .circleci/config.yml:150]: Why was this removed?

**Review by mdebarros — COMMENTED** [review]
See my comments

File: .ncurc.yml (hunk 1 above)
Comment (kleyow) [reply .ncurc.yml:1]: I thought we were using `.js` as the standard filetype, am I mistaken? What filetype are we setting on?

File: .circleci/config.yml (hunk 3 above)
Comment (kleyow) [reply .circleci/config.yml:150]: Oh I think I got confused accross my editor and believed that here doesn't have integration tests and accidently removed it.

File: package.json (hunk 2 above)
Comment (kleyow) [reply package.json:121]: done.

File: .ncurc.yml (hunk 1 above)
Comment (kleyow) [reply .ncurc.yml:1]: Seems I'm mistaken :thinking: wonder where I got js into my head.

**Comment by kleyow** [comment]
```
Test Suite:GP Tests
Environment:Development
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 2573                          │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 2573                          │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 587                           │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 129                           │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Thu, 14 Jul 2022 22:05:11 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Thu, 14 Jul 2022 22:08:06 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 175246 ms                     │
└───────────────────┴───────────────────────────────┘
TTK-Assertion-Report-multi-2022-07-14T22:08:06.506Z.html was generated
Terminate with exit code 0
Done

Test Suite:Bulk Tests
Environment:Development
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 183                           │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 183                           │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 17                            │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 4                             │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Thu, 14 Jul 2022 22:08:15 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Thu, 14 Jul 2022 22:08:38 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 23563 ms                      │
└───────────────────┴───────────────────────────────┘
TTK-Assertion-Report-multi-2022-07-14T22:08:38.861Z.html was generated
Terminate with exit code 0
Done
```

**Review by elnyry-sam-k — APPROVED** [review]

**Review by koekiebox — COMMENTED** [review]
:+1:
